src/ctapipe/reco/stereo_combination.py

```diff
@@ -404,3 +478,313 @@ def predict_table(self, mono_predictions: Table) -> Table:
+                        f"No valid DISP reconstruction parameter found for "
+                        f"prefix='{self.prefix}'). "
+                        f"Make sure to apply the DispReconstructor before using the "
+                        f"StereoDispCombiner or adapt the prefix accordingly."
+                    )
+
+                sign_score = dl2.disp[self.prefix].sign_score
+                dist_weight = self._calc_dist_weights(disp, sign_score, signs)
+                dist_weights.append(dist_weight)
+
+                abs_disp = np.abs(disp)
+                fov_lons = hillas_fov_lon + signs * abs_disp * np.cos(hillas_psi)
+                fov_lats = hillas_fov_lat + signs * abs_disp * np.sin(hillas_psi)
+                fov_lon_values.append(fov_lons)
+                fov_lat_values.append(fov_lats)
+                weights.append(self._calculate_weights(dl1) if dl1 else 1)
+                ids.append(tel_id)
+
+        if len(fov_lon_values) > 1:
+            index_tel_combs = get_combinations(len(ids), self.n_tel_combinations)
+            fov_lons, fov_lats, comb_weights = calc_combs_min_distances_event(
+                index_tel_combs,
+                np.array(fov_lon_values),
+                np.array(fov_lat_values),
+                np.array(weights),
+                np.array(dist_weights),
+            )
+            fov_lon_weighted_average = np.average(fov_lons, weights=comb_weights)
+            fov_lat_weighted_average = np.average(fov_lats, weights=comb_weights)
+            valid = True
+
+        # single tel events
+        elif len(fov_lon_values) == 1:
+            fov_lon_weighted_average = hillas_fov_lon + disp * np.cos(hillas_psi)
+            fov_lat_weighted_average = hillas_fov_lat + disp * np.sin(hillas_psi)
+            valid = True
+
+        else:
+            alt = az = u.Quantity(np.nan, u.deg, copy=COPY_IF_NEEDED)
+            valid = False
+
+        if valid:
+            alt, az = telescope_to_horizontal(
+                lon=fov_lon_weighted_average * u.deg,
+                lat=fov_lat_weighted_average * u.deg,
+                pointing_alt=event.monitoring.pointing.array_altitude,
+                pointing_az=event.monitoring.pointing.array_azimuth,
+            )
+
+        event.dl2.stereo.geometry[self.prefix] = ReconstructedGeometryContainer(
+            alt=alt,
+            az=az,
+            telescopes=ids,
+            is_valid=valid,
+            prefix=self.prefix,
+        )
+
+    def __call__(self, event: ArrayEventContainer) -> None:
+        """
+        Perform DISP-based stereo direction reconstruction for a single event.
+
+        This is the entry point used for event-wise processing.
+        Calls :meth:`_combine_altaz` and stores the resulting stereo geometry
+        inside ``event.dl2.stereo`` under the configured prefix.
+        """
+
+        self._combine_altaz(event)
+
+    def predict_table(self, mono_predictions: Table) -> Table:
```

Comment:
How much slower, if at all, is using `predict_table` for a single subarray event?

I'm asking because it would be nice if we could just use `predict_table` also for the `__call__` to reduce the code duplication and chance for behavorial differences between the container and the table case.